static/app/components/pluginConfig.tsx

```diff
@@ -46,14 +46,14 @@ class PluginConfig extends Component<Props, State> {
     this.loadPlugin(this.props.data);
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps: Props) {
-    this.loadPlugin(nextProps.data);
-  }
-
   shouldComponentUpdate(nextProps: Props, nextState: State) {
     return !isEqual(nextState, this.state) || !isEqual(nextProps.data, this.props.data);
   }
 
+  componentDidUpdate() {
+    this.loadPlugin(this.props.data);
+  }
```

Comment:
Right? I guess before it wouldn't react to state 🤔 maybe this breaks everything

Reply:
You should check the loading state and not have it call again, or check `plugins.isLoaded(this.props.data)`. There is no 'already loaded' check baked into the `load` in the Registry from what I can see.